packages/perseus/src/__tests__/widgets.test.ts

```diff
@@ -1,13 +1,107 @@
-import {testDependencies} from "../../../../testing/test-dependencies";
-import * as Dependencies from "../dependencies";
+import {mockStrings} from "../strings";
 import {registerAllWidgetsForTesting} from "../util/register-all-widgets-for-testing";
+import {generateTestRadioWidget} from "../util/test-utils";
 import * as Widgets from "../widgets";
 
 describe("Widget API support", () => {
     beforeAll(() => {
         registerAllWidgetsForTesting();
     });
 
+    describe("replaceWidget", () => {
+        it("replaces an existing widget", () => {
+            Widgets.replaceWidget("transformer", "radio");
+            expect(Widgets.getWidget("transformer")?.name).toBe("Radio");
+        });
+
+        it("Throws when the replacement isn't available", () => {
+            expect(() => Widgets.replaceWidget("radio", "dog-cat")).toThrow();
+        });
+    });
+
+    describe("getTransform", () => {
```

Comment:
Most of the test changes in this file are because I wanted to get rid of `underscore`; I prefer to have tests to work against to make sure I don't regress.